app/controllers/upload_controller.rb

```diff
@@ -114,6 +117,14 @@ def find_uploaders
                      sort_by(&:name)
   end
 
+  def validate_state
+    if Issue.states.keys.include?(params[:state]) || @uploader.to_s.include?('::Projects')
+      @state = params[:state]
+    else
```

Comment:
Can you review this logic? It seems that if I choose that uploader then I can pass anything I want to params[:state]